File: DSA/Animesh_jha_590016813/day_85_question_2.py
def dfs_traversal(vertices, edges, start=0):
    graph = [[] for _ in range(vertices)]
    for u, v in edges:
        graph[u].append(v)
        graph[v].append(u)

    visited = [False] * vertices
    dfs_order = []

    def dfs(node):
        visited[node] = True
        dfs_order.append(node)
        for neighbor in graph[node]:
            if not visited[neighbor]:
                dfs(neighbor)

    dfs(start)
    return dfs_order

print("DFS Example 1:", dfs_traversal(5, [[0,1],[0,2],[1,3],[2,4]]))  
print("DFS Example 2:", dfs_traversal(4, [[0,1],[1,2],[2,3]]))

File: DSA/VaniJain_590012020/Day85Question2.c
#include <stdio.h>
#include <stdlib.h>
#define MAX 100
int adj[MAX][MAX];
int visited[MAX];
int n;
void DFS(int node) 
{
    visited[node] = 1;
    printf("%d ", node);
    for (int i = 0; i < n; i++) 
    {
        if (adj[node][i] && !visited[i]) 
        {
            DFS(i);
        }
    }
}
int main() 
{
    int e;
    printf("Enter number of vertices: ");
    scanf("%d", &n);
    printf("Enter number of edges: ");
    scanf("%d", &e);
    for (int i = 0; i < n; i++) 
    {
        visited[i] = 0;
        for (int j = 0; j < n; j++) 
        {
            adj[i][j] = 0;
        }
    }
    printf("Enter edges (u v) format:\n");
    for (int i = 0; i < e; i++) 
    {
        int u, v;
        scanf("%d %d", &u, &v);
        adj[u][v] = 1;
        adj[v][u] = 1; 
    }
    printf("DFS traversal starting from node 0: ");
    DFS(0);
    printf("\n");
    return 0;
}
